### DISCUSSION TITLE
编译报错，貌似是adg的问题

### DISCUSSION THREAD

**Discussion by ghost** [Boos4721/openwrt#117] [Q&A/答疑解惑] [answered]
小白一个，刚接触本地编译，昨晚尝试编译了一下，报错，貌似是adguardhome的原因，请求解答

make[3]: Entering directory '/home/axbng/openwrt/feeds/packages/net/adguardhome'
rm -rf /home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.pkgdir/adguardhome.installed /home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.pkgdir/adguardhome
mkdir -p /home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.pkgdir/adguardhome
CONFIG_GOLANG_MOD_CACHE_WORLD_READABLE="" GO_BUILD_CACHE_DIR="/home/axbng/openwrt/tmp/go-build" GO_MOD_CACHE_DIR="/home/axbng/openwrt/dl/go-mod-cache" GO_MOD_ARGS="-modcacherw" GO_PKG="github.com/AdguardTeam/AdGuardHome" GO_INSTALL_EXTRA="" GO_INSTALL_ALL="" GO_SOURCE_ONLY="" GO_BUILD_PKG="github.com/AdguardTeam/AdGuardHome" GO_EXCLUDES="" GO_GO_GENERATE="" GO_INSTALL_BIN_PATH="/usr/bin" BUILD_DIR="/home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6" GO_BUILD_DIR="/home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.go_work/build" GO_BUILD_BIN_DIR="/home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.go_work/build/bin/linux_arm64" GO_BUILD_DEPENDS_PATH="/usr/share/gocode" GO_BUILD_DEPENDS_SRC="/home/axbng/openwrt/staging_dir/target-aarch64_cortex-a53_musl/usr/share/gocode/src" /usr/bin/env bash ../../lang/golang//golang-build.sh install_bin "/home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.pkgdir/adguardhome"
install -d -m0755 /home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.pkgdir/adguardhome/etc/init.d
install -m0755 ./files/adguardhome.init /home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.pkgdir/adguardhome/etc/init.d/adguardhome
install: cannot stat './files/adguardhome.init': No such file or directory
make[3]: *** [Makefile:79: /home/axbng/openwrt/build_dir/target-aarch64_cortex-a53_musl/adguardhome-0.107.6/.pkgdir/adguardhome.installed] Error 1
make[3]: Leaving directory '/home/axbng/openwrt/feeds/packages/net/adguardhome'
time: package/feeds/packages/adguardhome/compile#0.58#0.61#0.75
    ERROR: package/feeds/packages/adguardhome failed to build.
make[2]: *** [package/Makefile:116: package/feeds/packages/adguardhome/compile] Error 1
make[2]: Leaving directory '/home/axbng/openwrt'
make[1]: *** [package/Makefile:110: /home/axbng/openwrt/staging_dir/target-aarch64_cortex-a53_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/axbng/openwrt'
make: *** [/home/axbng/openwrt/include/toplevel.mk:230：world] 错误 2

**Comment by he110777** [comment]
我也是移除adg之后就能正常编译

**Reply by ghost** [reply]
重新试一下吧，我这边好了

**Comment by Boos4721** [comment, marked as answer]
UPDATE FEEDS THEN CHECK IT

**Reply by ghost** [reply]
还是不行哦大佬，我甚至删了文件夹重新下载，但还是adguardhome报错。
它显示的版本是0.107.6，我看了下是5天前发布的，是不是源码哪里没有来得及更改，望排查，辛苦。

**Reply by Boos4721** [reply]
> 还是不行哦大佬，我甚至删了文件夹重新下载，但还是adguardhome报错。 它显示的版本是0.107.6，我看了下是5天前发布的，是不是源码哪里没有来得及更改，望排查，辛苦。

https://github.com/Boos4721/openwrt/discussions/102

**Reply by ghost** [reply]
又尝试了一下，这次成功编译，THX
